docs/deployment/gae.rst

```diff
@@ -1,225 +1,104 @@
+        :linenos:
 
-#. Hack on your pyramid application, using a normal run, debug, restart
-   process.  For tips on how to use the ``pdb`` module within Google
-   App Engine, `see this blog post
-   <http://jjinux.blogspot.com/2008/05/python-debugging-google-app-engine-apps.html>`_.
-   In particular, you can create a function like so and call it to
-   drop your console into a pdb trace::
+        Pyramid
+        waitress
+        pyramid_debugtoolbar
+        pyramid_chameleon
 
-      def set_trace():
-          import pdb, sys
-          debugger = pdb.Pdb(stdin=sys.__stdin__, 
-              stdout=sys.__stdout__)
-          debugger.set_trace(sys._getframe().f_back)
+#. Edit ``main.py``
 
-#. `Sign up for a GAE account <http://code.google.com/appengine/>`_
-   and create an application.  You'll need a mobile phone to accept an
-   SMS in order to receive authorization.
+   Add the following lines::
 
-#. Edit the application's ID in ``app.yaml`` to match the application
-   name you created during GAE account setup.
+        from pyramid.paster import get_app, setup_logging
+        ini_path = 'production.ini'
+        setup_logging(ini_path)
+        application = get_app(ini_path, 'main')
 
-   .. code-block:: yaml
+#. Edit ``appengine_config.py``
 
-      application: mycoolpyramidapp
+   Add the following lines::
 
-#. Upload the application
+        from google.appengine.ext import vendor
+        vendor.add('lib')
 
-   .. code-block:: text
+#. Edit ``app.yaml``
 
-      $ python2.5 /usr/local/bin/appcfg.py update pyramidapp/app
+   Add the following lines:
 
-   You almost certainly won't hit the 3000-file GAE file number limit
-   when invoking this command.  If you do, however, it will look like
-   so:
+    .. code-block:: yaml
 
-   .. code-block:: text
+        application: application-id
+        version: version
+        runtime: python27
+        api_version: 1
+        threadsafe: false
 
-       HTTPError: HTTP Error 400: Bad Request
-       Rolling back the update.
-       Error 400: --- begin server output ---
-       Max number of files and blobs is 3000.
-       --- end server output ---
+        handlers:
+        - url: /static
+          static_dir: pyramid_project/static
+        - url: /.*
+          script: main.application
 
-   If you do experience this error, you will be able to get around
-   this by zipping libraries. You can use ``pip`` to create zipfiles
-   from packages.  See :ref:`pip_zip` for more information about this.
+    Replace `application` with your App Engine app ID, `version` with the version you want to deploy to, and `pyramid_project`
+    in the `static_dir` definition with the directory name above your static assets. If your static assets are in the root
+    directory, you can just put `static`
 
-   A successful upload looks like so:
+    For more details about app.yaml, see `app.yaml Reference <https://cloud.google.com/appengine/docs/standard/python/config/appref>`_.
 
```

Comment:
I think you need to dedent the above lines one space to align with the "A" in "Add the following lines:".